torch/csrc/jit/runtime/register_prim_ops.cpp

```diff
@@ -893,13 +900,16 @@ RegisterOperators reg(
      // results
      DEFINE_GENERIC_OP_WITH_COMPLEX(
          aten::pow,
-         static_cast<double>(pow(a, b)),
-         static_cast<double>(pow(a, b)),
+         static_cast<double>(powWrapper(a, b)),
+         static_cast<double>(powWrapper(a, b)),
          static_cast<c10::complex<double>>(pow(a, b)),
          float,
          float,
          complex),
-     DEFINE_INT_FLOAT_OP(aten::pow, pow(a, b), float),
+     DEFINE_INT_FLOAT_OP(
+         aten::pow,
+         static_cast<double>(powWrapper(a, b)),
+         float),
      DEFINE_FLOAT_COMPLEX_OP(aten::pow, pow(a, b), complex),
```

Comment:
I tested in Python that if a we have 0 ** complex, and the real part of the complex number is negative, it also raises an error, could we add the check for this case?

Reply:
Good call, let me add it.

Reply:
I think Python prevents 0 to the power of any complex number, not just those where the real part is negative. For example, `print(0 ** complex(10, 1))` and `print(0 ** complex(-10, 1))` both give error `ZeroDivisionError: 0.0 to a negative or complex power.`

PyTorch on the other hand, return `0j` for `0 ** complex(10, 1)` and `(inf+nanj)` for `0 ** complex(-10, 1)`.

I think the discussion is more broadly whether PyTorch should allow 0 being raised to a complex number, which I do not have a definitive answer to. Therefore, it may make sense to create a separate issue to track the zero base complex exponent inconsistency and limit the scope of this diff to real numbers.